### ISSUE TITLE
Lone surrogate U+D83D is not a scalar value

### ISSUE THREAD

**Issue by tuktun** [socketio/socket.io#2820]
### You want to:

* [x] report a *bug*
* [ ] request a *feature*
### Setup
- OS: Debian 7
- socket.io version:1.4.8

### Other information (e.g. stacktraces, related issues, suggestions how to fix)

App falling with this error: 
```
Error: Lone surrogate U+D83D is not a scalar value
    at Error (native)
    at checkScalarValue (/bot/node_modules/socket.io/node_modules/engine.io/node_modules/engine.io-parser/node_modules/utf8/utf8.js:69:10)
    at encodeCodePoint (/bot/node_modules/socket.io/node_modules/engine.io/node_modules/engine.io-parser/node_modules/utf8/utf8.js:90:4)
    at Object.utf8encode [as encode] (/bot/node_modules/socket.io/node_modules/engine.io/node_modules/engine.io-parser/node_modules/utf8/utf8.js:111:18)
    at Object.exports.encodePacket (/bot/node_modules/socket.io/node_modules/engine.io/node_modules/engine.io-parser/lib/index.js:75:34)
    at encodeOne (/bot/node_modules/socket.io/node_modules/engine.io/node_modules/engine.io-parser/lib/index.js:385:13)
    at eachWithIndex (/bot/node_modules/socket.io/node_modules/engine.io/node_modules/engine.io-parser/lib/index.js:234:5)
    at map (/bot/node_modules/socket.io/node_modules/engine.io/node_modules/engine.io-parser/lib/index.js:241:5)
    at Object.exports.encodePayloadAsBinary (/bot/node_modules/socket.io/node_modules/engine.io/node_modules/engine.io-parser/lib/index.js:409:3)
    at Object.exports.encodePayload (/bot/node_modules/socket.io/node_modules/engine.io/node_modules/engine.io-parser/lib/index.js:203:20)
    at XHR.Polling.send (/bot/node_modules/socket.io/node_modules/engine.io/lib/transports/polling.js:246:10)
    at Socket.flush (/bot/node_modules/socket.io/node_modules/engine.io/lib/socket.js:416:20)
    at Socket.sendPacket (/bot/node_modules/socket.io/node_modules/engine.io/lib/socket.js:392:10)
    at Socket.send.Socket.write (/bot/node_modules/socket.io/node_modules/engine.io/lib/socket.js:353:8)
    at writeToEngine (/bot/node_modules/socket.io/lib/client.js:148:17)
    at Client.packet (/bot/node_modules/socket.io/lib/client.js:159:7)
    at Socket.packet (/bot/node_modules/socket.io/lib/socket.js:210:15)
    at /bot/node_modules/socket.io/node_modules/socket.io-adapter/index.js:143:30
    at Encoder.encode (/bot/node_modules/socket.io/node_modules/socket.io-adapter/node_modules/socket.io-parser/index.js:135:5)
    at Adapter.broadcast (/bot/node_modules/socket.io/node_modules/socket.io-adapter/index.js:121:16)
    at Namespace.emit (/bot/node_modules/socket.io/lib/namespace.js:219:18)
    at Server.(anonymous function) [as emit] (/bot/node_modules/socket.io/lib/index.js:369:16)
```

**Comment by darrachequesne** [comment]
This was:
- fixed by https://github.com/socketio/engine.io-parser/pull/68, released as engine.io-parser `1.3.0`
- included in engine.io with https://github.com/socketio/engine.io/pull/425 and released as `1.7.0`
- and finally included in socket.io with https://github.com/socketio/socket.io/pull/2707, included as `1.5.0`

**Comment by tuktun** [comment]
Okay, thank you for your answer :)
Closed
